flaskr/api.py

```diff
@@ -2,62 +2,90 @@
     def get(self):
-        return {'hello': 'world'}
+        return { 'hello': 'world' }
 
 class FileUploads(Resource):
+
     def __init__(self):
         self.parser = reqparse.RequestParser()
 
     def get(self):
         # get all files in the ./static/uploads directory
-        files = [f for f in os.listdir(UPLOAD_DIR)]
+        files = [f for f in os.listdir(UPLOAD_DIR + '/dicoms')]
         return {
-                'dir' : UPLOAD_DIR, 
+                'pathRelative': DICOM_PATH_RELATIVE,
+                'pathAbsolute': DICOM_PATH_ABSOLUTE,
                 'count' : len(files),
                 'files' : files,
                 }
 
     def post(self):
-        self.parser.add_argument('startComputation', required=True, type=str)
-        self.parser.add_argument('CTFile', required=False, \
-                type=werkzeug.datastructures.FileStorage, location='files')
-        self.parser.add_arguemnt('DoseDataFile', required=False, \
-                type=werkzeug.datastructures.FileStorage, location='files')
-        self.parser.add_argument('RSFile', required=False, \
+        self.parser.add_argument('dicomArchive', required=True, \
                 type=werkzeug.datastructures.FileStorage, location='files')
 
         args = self.parser.parse_args()
 
-        if 'CTFile' in args:
-            save_file(args.get('CTFile'), 'CT.zip')
-        if 'DoseSumFile' in args:
-            save_file(args.get('DoseSumFile'), 'DoseSum.zip')
-        if 'RSFile' in args:
-            save_file(args.get('RSFile'), 'RS.zip')
-        
-        if args['startComputation']:
-            payload = {
-                    'CT_fname': 'CT.zip',
-                    'DoseSum_fname': 'DoseSum.zip',
-                    'RS_fname': 'RS.zip'
-                    }
-            response = requests.request('GET', GENIE_API+'load', headers={}, data=payload, files=[])
-            return {
-                    'successful': True,
-                    'computationStarted': True,
+        # save the .zip archive
+        archive = args.get('dicomArchive')
+        archive.save(os.path.join(UPLOAD_DIR, archive.filename))
+
+        # unzip the archive
+        with zipfile.ZipFile(os.path.join(UPLOAD_DIR, archive.filename), 'r') as zip_ref, err:
+            if err == zipfile.BadZipFile:
+                handle_bad_zip_file(err)
+            elif err == zipfile.LargeZipFile:
+                handle_large_zip_file(err)
+            else:
+               zip_ref.extractall(os.path.join(UPLOAD_DIR, 'dicoms'))
+
+        # verify that the archive contains dicom files   
+        try:
+            check_archive_contents(path_to_archive)
+        except dicomutils.InvalidDicomName as err:
+            handle_invalid_contents(err)
+
+        return { 
+                'status': 'success',
+                'message': 'files saved to path',
+                'pathRelative': DICOM_PATH_RELATIVE,
+                'pathAbsolute': DICOM_PATH_ABSOLUTE,
```

Comment:
Just as small note for the future:
if you do want to use it as an argument to another endpoint, check if path, is containing this absoulte path. Otherwise, you are making our application vunerable to some kind of this [attack](https://owasp.org/www-project-top-ten/2017/A5_2017-Broken_Access_Control), where a user will be able to walk through the system by modyfiyng a query string.


Btw, why do you return an absolute and a relative path?

Reply:
As we discussed, it will be used in the Julia computation API. The relative path is unnecessary and will be removed